src/schedule/zephyr_domain.c

```diff
@@ -195,6 +196,8 @@ struct ll_schedule_domain *timer_domain_init(struct timer *timer, int clk)
 
 	zephyr_domain->ll_timer = timer;
 	zephyr_domain->ll_domain = domain;
+	/* 10 is rather random, we better not accumulate 10 missed timer interrupts */
+	k_sem_init(&zephyr_domain->sem, 0, 10);
```

Comment:
Can we make 10 a macro. This should bail out in 10 ms - Btw, where do we check this ?

Reply:
@lgirdwood no, this isn't 10ms, these are 10 misses. The maximum semaphore count. I.e. the timer interrupt increments the semaphore counter and wakes up the thread. Normally the thread wakes up immediately and decrements the counter back to 0. But sometimes the timer IRQ "gives" the semaphore before the thread suspends on it. Then the thread checks the semaphore when going to sleep and doesn't even suspend. In bad cases we can miss not 1 but several IRQs, then the timer interrupt will keep incrementing the counter until the thread hopefully manages to catch up and decrement it back to 0. So, in ideal case it would be `k_sem_init(&zephyr_domain->sem, 0, 1);` but we might as well add some security margin there... 2? 4? 10? Yes, this should be a macro, but maybe 10 was a bit too much...

Reply:
I think this is ok for the IRQ and we should add some trace to the scheduled thread and print overshoot warning there. The agent already emits a trace in this scenario... it notices the LL task scheduling is falling behind and emits a trace. But we could print a more explicit trace in the scheduler thread.

Reply:
@kv2019i sure, let's do this in an incremental update, we might want to polish that for a while - choose the context, the wording, etc. Would be good to merge the fix sooner